src/main/scala/com/gossiperl/client/GossiperlClient.scala

```diff
@@ -1,8 +1,36 @@
 package com.gossiperl.client
 
-trait GossiperlClient {
+import akka.actor._
+import akka.util.Timeout
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import concurrent.duration._
+import scala.concurrent.Promise
+import scala.util.{Failure, Success}
 
-  def withOverlay():Unit = {
+trait GossiperlClient extends LazyLogging {
+
+  val system = ActorSystem("gossiperl-system")
+  val supervisor = system.actorOf(Props[Supervisor], name=Supervisor.actorName)
+  import scala.concurrent.ExecutionContext.Implicits.global
+  implicit val timeout = Timeout(1 seconds)
+
+  def withOverlay( configuration: OverlayConfiguration ) ( action: Option[GossiperlProxy] => Unit ):Unit = {
+    system.actorSelection(s"/user/${Supervisor.actorName}").resolveOne( timeout.duration ) onComplete {
+      case Success(a) =>
+        val p = Promise[GossiperlProxy]
+        p.future onComplete {
+          case Success( proxy ) =>
+            action.apply( Some( proxy ) )
+          case Failure( ex ) =>
+            logger.error(s"Error while awaiting for gossiperl proxy with configuration $configuration.", ex)
+            action.apply( None )
+        }
+        a ! SupervisorProtocol.Connect( configuration, p )
```

Comment:
personally I'm not a fan of passing on the responsibility of completing a promise outside the current scope. Basically you're using promises to avoid having to send back a message from the actor. Also, by having the actor complete the promise, you can't be sure anymore that everything which happens in the actor happens in one single thread, which defeats one of the core principles of an actor.
I would just have the actor respond with a message, handle the response here and based on that either complete or fail the promise.